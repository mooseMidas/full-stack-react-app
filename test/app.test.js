// Test to check if API response

// supertest library installed
const request = require("supertest");
const app = require("../app");

describe("GET /search", () => {
  //api restricted to one request every 5 sec
  jest.setTimeout(10000);
  // to check if the endpoint returns results when given query parameters
  it("returns data from the iTunes API", async () => {
    const response = await request(app).get(
      "/search?term=apple&limit=10&entity=musicTrack"
    );

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});

/*
API Testing using Jest and SuperTest, Testing with Marie, Retrieved on 18 July 2023 from https://www.testingwithmarie.com/post/api-testing-using-jest-and-supertest
 */
