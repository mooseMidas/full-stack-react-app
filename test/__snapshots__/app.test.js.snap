// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /search returns data from the iTunes API 1`] = `
{
  "resultCount": 10,
  "results": [
    {
      "artistId": 255303209,
      "artistName": "Flo Rida",
      "artistViewUrl": "https://music.apple.com/us/artist/flo-rida/255303209?uo=4",
      "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music125/v4/14/42/a0/1442a084-36ea-09a0-ff87-d5aea0cc19f8/mzi.ktgesuhn.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music125/v4/14/42/a0/1442a084-36ea-09a0-ff87-d5aea0cc19f8/mzi.ktgesuhn.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music125/v4/14/42/a0/1442a084-36ea-09a0-ff87-d5aea0cc19f8/mzi.ktgesuhn.jpg/60x60bb.jpg",
      "collectionCensoredName": "Mail On Sunday",
      "collectionExplicitness": "explicit",
      "collectionId": 275496110,
      "collectionName": "Mail On Sunday",
      "collectionPrice": 9.99,
      "collectionViewUrl": "https://music.apple.com/us/album/low-feat-t-pain/275496110?i=275496141&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview122/v4/29/cd/85/29cd85b5-ebc0-e723-25bf-a327a7f9aee8/mzaf_10292750635246506192.plus.aac.p.m4a",
      "primaryGenreName": "Hip-Hop/Rap",
      "releaseDate": "2007-10-09T07:00:00Z",
      "trackCensoredName": "Low (feat. T-Pain)",
      "trackCount": 14,
      "trackExplicitness": "notExplicit",
      "trackId": 275496141,
      "trackName": "Low (feat. T-Pain)",
      "trackNumber": 5,
      "trackPrice": 1.29,
      "trackTimeMillis": 231400,
      "trackViewUrl": "https://music.apple.com/us/album/low-feat-t-pain/275496110?i=275496141&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 316069,
      "artistName": "Alicia Keys & Pharrell Williams",
      "artistViewUrl": "https://music.apple.com/us/artist/alicia-keys/316069?uo=4",
      "artworkUrl100": "https://is1-ssl.mzstatic.com/image/thumb/Music125/v4/77/ea/82/77ea8270-effd-fd43-d751-556f43e13306/886446217161.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is1-ssl.mzstatic.com/image/thumb/Music125/v4/77/ea/82/77ea8270-effd-fd43-d751-556f43e13306/886446217161.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is1-ssl.mzstatic.com/image/thumb/Music125/v4/77/ea/82/77ea8270-effd-fd43-d751-556f43e13306/886446217161.jpg/60x60bb.jpg",
      "collectionArtistId": 80204262,
      "collectionArtistName": "Various Artists",
      "collectionCensoredName": "Hidden Figures: The Album",
      "collectionExplicitness": "notExplicit",
      "collectionId": 1173993509,
      "collectionName": "Hidden Figures: The Album",
      "collectionPrice": 9.99,
      "collectionViewUrl": "https://music.apple.com/us/album/apple/1173993509?i=1173993765&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/cc/8f/67/cc8f678f-d46b-547c-0166-18f61207ba45/mzaf_15258899724048646454.plus.aac.p.m4a",
      "primaryGenreName": "Soundtrack",
      "releaseDate": "2016-12-09T12:00:00Z",
      "trackCensoredName": "Apple",
      "trackCount": 9,
      "trackExplicitness": "notExplicit",
      "trackId": 1173993765,
      "trackName": "Apple",
      "trackNumber": 6,
      "trackPrice": 1.29,
      "trackTimeMillis": 227144,
      "trackViewUrl": "https://music.apple.com/us/album/apple/1173993509?i=1173993765&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 378430797,
      "artistName": "Julia Michaels",
      "artistViewUrl": "https://music.apple.com/us/artist/julia-michaels/378430797?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music122/v4/9a/2e/19/9a2e192c-903d-02aa-ce74-8dd8f5329fb1/19UMGIM03027.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music122/v4/9a/2e/19/9a2e192c-903d-02aa-ce74-8dd8f5329fb1/19UMGIM03027.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music122/v4/9a/2e/19/9a2e192c-903d-02aa-ce74-8dd8f5329fb1/19UMGIM03027.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Inner Monologue, Pt. 1 - EP",
      "collectionExplicitness": "explicit",
      "collectionId": 1449932668,
      "collectionName": "Inner Monologue, Pt. 1 - EP",
      "collectionPrice": 5.99,
      "collectionViewUrl": "https://music.apple.com/us/album/apple/1449932668?i=1449932877&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview122/v4/23/bb/4a/23bb4a7a-038d-76bd-2516-9ca0b3bcb2cf/mzaf_2022759717471534732.plus.aac.p.m4a",
      "primaryGenreName": "Pop",
      "releaseDate": "2019-01-24T12:00:00Z",
      "trackCensoredName": "Apple",
      "trackCount": 6,
      "trackExplicitness": "notExplicit",
      "trackId": 1449932877,
      "trackName": "Apple",
      "trackNumber": 5,
      "trackPrice": 1.29,
      "trackTimeMillis": 164323,
      "trackViewUrl": "https://music.apple.com/us/album/apple/1449932668?i=1449932877&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 1646302,
      "artistName": "The Smashing Pumpkins",
      "artistViewUrl": "https://music.apple.com/us/artist/the-smashing-pumpkins/1646302?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Rotten Apples: Greatest Hits",
      "collectionExplicitness": "explicit",
      "collectionId": 712732565,
      "collectionName": "Rotten Apples: Greatest Hits",
      "collectionPrice": 7.99,
      "collectionViewUrl": "https://music.apple.com/us/album/1979/712732565?i=712732634&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/33/ef/da/33efdabd-d35b-4419-a1a3-e04fb7a238dd/mzaf_17748139787921487821.plus.aac.p.m4a",
      "primaryGenreName": "Hard Rock",
      "releaseDate": "1995-10-23T07:00:00Z",
      "trackCensoredName": "1979",
      "trackCount": 18,
      "trackExplicitness": "notExplicit",
      "trackId": 712732634,
      "trackName": "1979",
      "trackNumber": 9,
      "trackPrice": 1.29,
      "trackTimeMillis": 263573,
      "trackViewUrl": "https://music.apple.com/us/album/1979/712732565?i=712732634&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 1646302,
      "artistName": "The Smashing Pumpkins",
      "artistViewUrl": "https://music.apple.com/us/artist/the-smashing-pumpkins/1646302?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Rotten Apples: Greatest Hits",
      "collectionExplicitness": "explicit",
      "collectionId": 712732565,
      "collectionName": "Rotten Apples: Greatest Hits",
      "collectionPrice": 7.99,
      "collectionViewUrl": "https://music.apple.com/us/album/bullet-with-butterfly-wings/712732565?i=712732631&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/4c/bc/4f/4cbc4fc5-fc8c-b159-9c78-c8c91ecf2c94/mzaf_1762488723687916470.plus.aac.p.m4a",
      "primaryGenreName": "Hard Rock",
      "releaseDate": "1995-10-24T12:00:00Z",
      "trackCensoredName": "Bullet with Butterfly Wings",
      "trackCount": 18,
      "trackExplicitness": "notExplicit",
      "trackId": 712732631,
      "trackName": "Bullet with Butterfly Wings",
      "trackNumber": 8,
      "trackPrice": 1.29,
      "trackTimeMillis": 257227,
      "trackViewUrl": "https://music.apple.com/us/album/bullet-with-butterfly-wings/712732565?i=712732631&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 1646302,
      "artistName": "The Smashing Pumpkins",
      "artistViewUrl": "https://music.apple.com/us/artist/the-smashing-pumpkins/1646302?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Rotten Apples: Greatest Hits",
      "collectionExplicitness": "explicit",
      "collectionId": 712732565,
      "collectionName": "Rotten Apples: Greatest Hits",
      "collectionPrice": 7.99,
      "collectionViewUrl": "https://music.apple.com/us/album/today/712732565?i=712732583&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/a2/ee/83/a2ee8356-11b1-caff-1700-d8d2f1158af7/mzaf_14918469876557823103.plus.aac.p.m4a",
      "primaryGenreName": "Hard Rock",
      "releaseDate": "1993-07-26T12:00:00Z",
      "trackCensoredName": "Today",
      "trackCount": 18,
      "trackExplicitness": "notExplicit",
      "trackId": 712732583,
      "trackName": "Today",
      "trackNumber": 5,
      "trackPrice": 1.29,
      "trackTimeMillis": 202133,
      "trackViewUrl": "https://music.apple.com/us/album/today/712732565?i=712732583&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 1646302,
      "artistName": "The Smashing Pumpkins",
      "artistViewUrl": "https://music.apple.com/us/artist/the-smashing-pumpkins/1646302?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Rotten Apples: Greatest Hits",
      "collectionExplicitness": "explicit",
      "collectionId": 712732565,
      "collectionName": "Rotten Apples: Greatest Hits",
      "collectionPrice": 7.99,
      "collectionViewUrl": "https://music.apple.com/us/album/disarm/712732565?i=712732586&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/98/94/3d/98943d2a-7ba4-c901-4884-aafa2bdb8aa0/mzaf_3776415201077849967.plus.aac.p.m4a",
      "primaryGenreName": "Hard Rock",
      "releaseDate": "1993-07-27T12:00:00Z",
      "trackCensoredName": "Disarm",
      "trackCount": 18,
      "trackExplicitness": "notExplicit",
      "trackId": 712732586,
      "trackName": "Disarm",
      "trackNumber": 6,
      "trackPrice": 1.29,
      "trackTimeMillis": 198467,
      "trackViewUrl": "https://music.apple.com/us/album/disarm/712732565?i=712732586&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 1646302,
      "artistName": "The Smashing Pumpkins",
      "artistViewUrl": "https://music.apple.com/us/artist/the-smashing-pumpkins/1646302?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Rotten Apples: Greatest Hits",
      "collectionExplicitness": "explicit",
      "collectionId": 712732565,
      "collectionName": "Rotten Apples: Greatest Hits",
      "collectionPrice": 7.99,
      "collectionViewUrl": "https://music.apple.com/us/album/tonight-tonight/712732565?i=712732637&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/fa/25/bd/fa25bdc1-e218-42c8-d85d-146ed9604772/mzaf_14820100357801315685.plus.aac.p.m4a",
      "primaryGenreName": "Hard Rock",
      "releaseDate": "1995-10-23T12:00:00Z",
      "trackCensoredName": "Tonight, Tonight",
      "trackCount": 18,
      "trackExplicitness": "notExplicit",
      "trackId": 712732637,
      "trackName": "Tonight, Tonight",
      "trackNumber": 11,
      "trackPrice": 1.29,
      "trackTimeMillis": 255600,
      "trackViewUrl": "https://music.apple.com/us/album/tonight-tonight/712732565?i=712732637&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 160032,
      "artistName": "Cibo Matto",
      "artistViewUrl": "https://music.apple.com/us/artist/cibo-matto/160032?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/e0/d9/10/e0d910f1-fbc6-ea1b-f59f-83a4746c9f9c/mzi.ukmnzmcz.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/e0/d9/10/e0d910f1-fbc6-ea1b-f59f-83a4746c9f9c/mzi.ukmnzmcz.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/e0/d9/10/e0d910f1-fbc6-ea1b-f59f-83a4746c9f9c/mzi.ukmnzmcz.jpg/60x60bb.jpg",
      "collectionCensoredName": "Viva! La Woman",
      "collectionExplicitness": "explicit",
      "collectionId": 207955592,
      "collectionName": "Viva! La Woman",
      "collectionPrice": 9.99,
      "collectionViewUrl": "https://music.apple.com/us/album/apple/207955592?i=207955624&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview122/v4/f7/d3/e8/f7d3e8a8-8188-5660-51d5-ede536c0da4a/mzaf_17056883522469130290.plus.aac.p.m4a",
      "primaryGenreName": "Rock",
      "releaseDate": "1996-01-12T12:00:00Z",
      "trackCensoredName": "Apple",
      "trackCount": 11,
      "trackExplicitness": "notExplicit",
      "trackId": 207955624,
      "trackName": "Apple",
      "trackNumber": 1,
      "trackPrice": 1.29,
      "trackTimeMillis": 241280,
      "trackViewUrl": "https://music.apple.com/us/album/apple/207955592?i=207955624&uo=4",
      "wrapperType": "track",
    },
    {
      "artistId": 1646302,
      "artistName": "The Smashing Pumpkins",
      "artistViewUrl": "https://music.apple.com/us/artist/the-smashing-pumpkins/1646302?uo=4",
      "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/100x100bb.jpg",
      "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/30x30bb.jpg",
      "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/61/a5/ac/61a5ac46-0fd4-3fa0-7b3a-2993b2390b76/13UABIM56696.rgb.jpg/60x60bb.jpg",
      "collectionCensoredName": "Rotten Apples: Greatest Hits",
      "collectionExplicitness": "explicit",
      "collectionId": 712732565,
      "collectionName": "Rotten Apples: Greatest Hits",
      "collectionPrice": 7.99,
      "collectionViewUrl": "https://music.apple.com/us/album/cherub-rock/712732565?i=712732580&uo=4",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "isStreamable": true,
      "kind": "song",
      "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/85/5c/98/855c9866-814b-427f-4524-c28c0c82078b/mzaf_5592640880113573639.plus.aac.p.m4a",
      "primaryGenreName": "Hard Rock",
      "releaseDate": "1993-07-27T12:00:00Z",
      "trackCensoredName": "Cherub Rock",
      "trackCount": 18,
      "trackExplicitness": "notExplicit",
      "trackId": 712732580,
      "trackName": "Cherub Rock",
      "trackNumber": 4,
      "trackPrice": 1.29,
      "trackTimeMillis": 299293,
      "trackViewUrl": "https://music.apple.com/us/album/cherub-rock/712732565?i=712732580&uo=4",
      "wrapperType": "track",
    },
  ],
}
`;
